assets/js/theme/global/currency-selector.js

```diff
@@ -1,5 +1,57 @@
-export default function () {
+import swal from './sweet-alert';
+import utils from '@bigcommerce/stencil-utils';
+
+export default function (cartId) {
+    function changeCurrency(url, currencyCode) {
+        $.ajax({
+            url,
+            method: 'POST',
+            data: JSON.stringify({ currencyCode }),
+        }).then(() => {
+            window.location.reload();
+        });
+    }
+
     $(document.body).on('click', '.currencySelector', () => {
         $('.currency-selection-list').toggleClass('active');
     });
+
+    $(document.body).on('click', '.dropdown-menu-item a', event => {
+        if (!cartId) {
+            return;
+        }
+
+        event.preventDefault();
+
+        utils.api.cart.getCart({ cartId }, (err, response) => {
+            let showWarning = false;
+            response.discounts.forEach((discount) => {
+                if (discount > 0) {
+                    showWarning = true;
+                }
+            });
+
+            if (response.coupons.length > 0) {
+                showWarning = true;
+            }
+
+            if (response.lineItems.giftCertificates.length > 0) {
+                showWarning = true;
+            }
+
+            if (showWarning) {
+                swal.fire({
+                    text: $(event.target).attr('data-warning'),
+                    icon: 'warning',
+                    showCancelButton: true,
+                }).then(result => {
+                    if (result.value && result.value === true) {
+                        changeCurrency($(event.target).attr('data-cart-currency-switch-url'), $(event.target).attr('data-currency-code'));
```

Comment:
I believe jQuery has a `data` method that can be used to retrieve `data-*` attributes. i.e.: `.data('cart-currency-switch-url')`.